### PR TITLE
Check the system xdg dir for a yab config file

### PR DESCRIPTION
**Pull request by eklitzke**
This tests the user's config dir for a yab config, and if that fails yab
will fallback to checking the system directories. On a Linux machine
this means that the config files will be checked in this order (with the
first found taking precedence):

    - ~/.config/yab/defaults.ini
    - /etc/xdg/yab/defaults.ini

The motivation behind this change is to allow system administrators to
install a system-wide default configuration file.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -206,10 +206,19 @@ func overrideDefaults(defaults *Options, args []string) {
 func parseDefaultConfigs(parser *flags.Parser) error {
 	app := xdg.App{Name: "yab"}
 
-	configFile := app.ConfigPath("defaults.ini")
-	if _, err := os.Stat(configFile); err != nil {
-		// No defaults file to read
-		return nil
+	// Find the best config path to use, preferring the user's config path and
```
Comment (prashantv) [inline main.go]: Can load multiple files? I imagine if there are multiple files, we'd try the system ones first, then override from the user's configuration?

Comment (eklitzke) [inline main.go]: AFAIK the current parser doesn't allow this (although obviously it's possible to do in principle).

**Comment by eklitzke** [comment]
OK, will add tests.

File: main.go (hunk 1 above)
Comment (prashantv) [inline main.go]: what happens if you call `iniPaser.ParseFile` multiple times?

Comment (eklitzke) [inline main.go]: Was just thinking about this. The way XDG works for all of the files on my Linux desktop is that it does not merge files, it uses the same precedence logic I've implemented here. The recommended way to customize a system XDG file on Linux is to copy the system one to your home directory and then modify it.

The issue here is that some of the flags we use with yab do something only if set. That means that you can't meaningfully unset a flag by putting it in your home directory and clearing that value.

**Comment by eklitzke** [comment]
Just added test, let me know what you think.

File: main.go (hunk 1 above)
Comment (prashantv) [inline main.go]: Ok cool, makes sense.

**Comment by prashantv** [comment]
lgtm!

**Pull request merged**